### PR TITLE
Enables the logger to handle std::string directly

### PR DESCRIPTION
**Pull request by luc-guyot-infomaniak**

### CODE REVIEW

File: src/libcommonserver/db/db.cpp [hunk 1]
```diff
@@ -376,7 +376,7 @@ bool Db::init(const std::string &version) {
             queryFree(CREATE_VERSION_TABLE_ID);
 
             // Insert version
-            LOG_DEBUG(_logger, "Insert version " << version.c_str());
+            LOG_DEBUG(_logger, "Insert version " << version);
```
Comment (luc-guyot-infomaniak) [inline src/libcommonserver/db/db.cpp:379]: Just to test an actual use case.

File: src/libcommonserver/db/db.cpp [hunk 2]
```diff
@@ -386,7 +386,7 @@ bool Db::init(const std::string &version) {
             queryFree(INSERT_VERSION_REQUEST_ID);
 
             // Create DB
-            LOG_INFO(_logger, "Create " << dbType().c_str() << " DB");
+            LOG_INFO(_logger, "Create " << dbType() << " DB");
```
Comment (luc-guyot-infomaniak) [inline src/libcommonserver/db/db.cpp:389]: Idem

File: src/libcommonserver/db/db.cpp [hunk 3]
```diff
@@ -413,7 +413,7 @@ bool Db::init(const std::string &version) {
 void Db::startTransaction() {
     if (!_transaction) {
         if (!_sqliteDb->startTransaction()) {
-            LOG_WARN(_logger, "ERROR starting transaction: " << _sqliteDb->error().c_str());
+            LOG_WARN(_logger, "ERROR starting transaction: " << _sqliteDb->error());
```
Comment (luc-guyot-infomaniak) [inline src/libcommonserver/db/db.cpp:416]: Idem

File: test/server/socketapi/testsocketapi.cpp [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    QByteArray b1 = localPath.toLatin1();
+    CPPUNIT_ASSERT(QString(b1) == localPath);
+
+    // QString parentPath = QFileInfo(localPath).dir().path().toUtf8();
+
+    // CPPUNIT_ASSERT(QString("/Users/hyphensmbp/kDrive Labör").toStdString() == parentPath.toStdString());
+
+    // std::filesystem::path localPath_ = QStr2Path(parentPath);
+
+    // CPPUNIT_ASSERT(localPath_ == SyncPath(QString("/Users/hyphensmbp/kDrive Labör").toStdString()));
```
Comment (ClementKunz) [inline test/server/socketapi/testsocketapi.cpp:49]: Remove unnecessary comments?

File: test/server/socketapi/testsocketapi.cpp [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+void TestSocketApi::setUp() {}
+
+void TestSocketApi::tearDown() {}
+
+void TestSocketApi::testFileData() {
+    FileData fileData;
+    QString localPath = QString("/Users/hyphensmbp/kDrive Labör/Common documents/");
+
+    QByteArray b1 = localPath.toLatin1();
+    CPPUNIT_ASSERT(QString(b1) == localPath);
```
[the remark this answers is not in the document]
Comment (luc-guyot-infomaniak) [reply test/server/socketapi/testsocketapi.cpp:41]: Thanks for catching this: this is an accidental commit, unrelated to the scope of this PR. To be removed entirely.

File: test/server/socketapi/testsocketapi.cpp (hunk 4 above)
Comment (luc-guyot-infomaniak) [reply test/server/socketapi/testsocketapi.cpp:49]: Idem. I'll remove the new test files.

File: test/server/socketapi/testsocketapi.cpp (hunk 5 above)
[the remark this answers is not in the document]
Comment (luc-guyot-infomaniak) [reply test/server/socketapi/testsocketapi.cpp]: Done.

**Pull request merged**